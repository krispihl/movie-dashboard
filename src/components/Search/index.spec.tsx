import { Search } from "./index";
import { render, screen, act } from '@testing-library/react';
import userEvent from "@testing-library/user-event";

const mockedMovies = [
    {
        adult: false,
        backdrop_path: '/backdrop.jpg',
        genre_ids: [16, 28],
        id: 123456,
        original_language: 'en',
        original_title: 'The Movie',
        overview: 'Story about writing tests',
        popularity: 500,
        poster_path: '/poster.jpg',
        release_date: '2023-03-07',
        title: 'The Movie',
        video: false,
        vote_average: 8.0,
        vote_count: 666,
        rowPosition: 0,
    }
];

jest.mock('../../utils/recursive-fetch', () => ({
    recursiveFetch: () => mockedMovies
}));

jest.mock('../../contexts/movies', () => ({
    useMovies: () => ({
        allMovies: mockedMovies,
        setFilteredMovies: jest.fn(),
    })
}));

describe('src/components/Search', () => {
    it('should render Search component correctly', async () => {
        const { container } = render(<Search />);

        expect(container).toMatchSnapshot();
        expect(screen.getByRole('textbox')).toBeInTheDocument();
        expect(document.querySelector('.material-icons')).toHaveTextContent('search');
    });

    it('should render Search component correctly when user has typed searchterm', async () => {
        let container;

        await act(async () => {
            container = render(<Search />).container;
        });

        act(() => {
            userEvent.type(screen.getByRole('textbox'), 'The');
        });

        expect(container).toMatchSnapshot();
        expect(document.querySelector('.material-icons')).toHaveTextContent('clear');
    });

    it('should clear searchterm when clicked on clear icon', async () => {
        let container;

        await act(async () => {
            container = render(<Search />).container;
        });

        act(() => {
            userEvent.type(screen.getByRole('textbox'), 'The');
        });

        const clearIcon = document.querySelector('.search__icon--clear');

        await act(async () => {
            if (clearIcon) {
                userEvent.click(clearIcon);
            }
        });

        expect(container).toMatchSnapshot();
        expect(screen.getByRole('textbox')).toHaveAttribute('value', '');
    });
});
