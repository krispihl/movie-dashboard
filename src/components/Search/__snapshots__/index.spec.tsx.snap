// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Search should clear searchterm when clicked on clear icon 1`] = `
<div>
  <div
    class="search"
  >
    <input
      class="search__input"
      placeholder="Search for movies..."
      value=""
    />
    <span
      class="material-icons search__icon"
    >
      search
    </span>
  </div>
</div>
`;

exports[`src/components/Search should render Search component correctly 1`] = `
<div>
  <div
    class="search"
  >
    <input
      class="search__input"
      placeholder="Search for movies..."
      value=""
    />
    <span
      class="material-icons search__icon"
    >
      search
    </span>
  </div>
</div>
`;

exports[`src/components/Search should render Search component correctly when user has typed searchterm 1`] = `
<div>
  <div
    class="search"
  >
    <input
      class="search__input"
      placeholder="Search for movies..."
      value="The"
    />
    <span
      class="material-icons search__icon search__icon--clear"
    >
      clear
    </span>
  </div>
</div>
`;
